client/src/ui/organisms/article-actions/bookmark-menu/menu-v2.tsx

```diff
@@ -82,23 +104,24 @@ export default function BookmarkV2Menu({ doc }: { doc: Doc }) {
 
   const cancelHandler = (e: React.MouseEvent) => {
     e.preventDefault();
-    setFormItem(savedItem || defaultItem);
+    setFormItem((savedItems && savedItems[0]) || defaultItem);
     setShow(false);
   };
 
+  const isCurrentInCollection = () =>
+    "id" in formItem &&
+    savedItems?.length &&
+    savedItems.some((item) => item.collection_id === formItem.collection_id);
+
   const saveHandler = async (
     e: React.FormEvent<HTMLFormElement> | React.BaseSyntheticEvent
   ) => {
     e.preventDefault();
     if (!collections || isPending) return;
     setLastAction("save");
     resetErrors();
-    if ("id" in formItem && savedItem) {
-      if (savedItem.collection_id !== formItem.collection_id) {
-        await Promise.all([deleteItem(savedItem), addItem(formItem)]);
-      } else {
-        await editItem(formItem);
-      }
+    if (isCurrentInCollection()) {
+      await editItem(formItem as Item);
```

Comment:
...and you don't need to force the type

```suggestion
    if ("id" in formItem && isCurrentInCollection()) {
      await editItem(formItem);
```